<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#INTRODUCTION" data-toc-modified-id="INTRODUCTION-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>INTRODUCTION</a></span></li><li><span><a href="#Libraries:-installation-and-importing" data-toc-modified-id="Libraries:-installation-and-importing-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Libraries: installation and importing</a></span><ul class="toc-item"><li><span><a href="#Exploratory-data-analysis-(EDA)" data-toc-modified-id="Exploratory-data-analysis-(EDA)-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Exploratory data analysis (EDA)</a></span><ul class="toc-item"><li><span><a href="#Import-data" data-toc-modified-id="Import-data-2.1.1"><span class="toc-item-num">2.1.1&nbsp;&nbsp;</span>Import data</a></span></li><li><span><a href="#Get-some-information-about-the-dataframe-(completeness,-data-types,-column-names)" data-toc-modified-id="Get-some-information-about-the-dataframe-(completeness,-data-types,-column-names)-2.1.2"><span class="toc-item-num">2.1.2&nbsp;&nbsp;</span>Get some information about the dataframe (completeness, data types, column names)</a></span></li><li><span><a href="#Get-some-statistical-information" data-toc-modified-id="Get-some-statistical-information-2.1.3"><span class="toc-item-num">2.1.3&nbsp;&nbsp;</span>Get some statistical information</a></span><ul class="toc-item"><li><span><a href="#Graph-to-show-outliers" data-toc-modified-id="Graph-to-show-outliers-2.1.3.1"><span class="toc-item-num">2.1.3.1&nbsp;&nbsp;</span>Graph to show outliers</a></span></li></ul></li><li><span><a href="#Get-all-possible-values-for-column-named-&quot;Country-Name&quot;-and-&quot;Country-code&quot;,-so-we-can-analyze-if-we-should-ignore-rows-(contries-in-this-case)" data-toc-modified-id="Get-all-possible-values-for-column-named-&quot;Country-Name&quot;-and-&quot;Country-code&quot;,-so-we-can-analyze-if-we-should-ignore-rows-(contries-in-this-case)-2.1.4"><span class="toc-item-num">2.1.4&nbsp;&nbsp;</span>Get all possible values for column named "Country Name" and "Country code", so we can analyze if we should ignore rows (contries in this case)</a></span></li></ul></li><li><span><a href="#Data-analysis" data-toc-modified-id="Data-analysis-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Data analysis</a></span><ul class="toc-item"><li><span><a href="#Descriptive-analytics:-graphs" data-toc-modified-id="Descriptive-analytics:-graphs-2.2.1"><span class="toc-item-num">2.2.1&nbsp;&nbsp;</span>Descriptive analytics: graphs</a></span><ul class="toc-item"><li><span><a href="#Bar-graphs" data-toc-modified-id="Bar-graphs-2.2.1.1"><span class="toc-item-num">2.2.1.1&nbsp;&nbsp;</span>Bar graphs</a></span></li><li><span><a href="#Pie-charts" data-toc-modified-id="Pie-charts-2.2.1.2"><span class="toc-item-num">2.2.1.2&nbsp;&nbsp;</span>Pie charts</a></span></li><li><span><a href="#World-map" data-toc-modified-id="World-map-2.2.1.3"><span class="toc-item-num">2.2.1.3&nbsp;&nbsp;</span>World map</a></span></li><li><span><a href="#Line-charts" data-toc-modified-id="Line-charts-2.2.1.4"><span class="toc-item-num">2.2.1.4&nbsp;&nbsp;</span>Line charts</a></span></li></ul></li></ul></li></ul></li></ul></div>

# INTRODUCTION

The visual representation of data is key to share and interpret information. Graphs are one of the most important elements in decision-making processes.
Here, you are going to have an example of different ways to show data and how different data can be interpreted based on the way it is presented.

# Libraries: installation and importing

If you don't have the needed libraries, you can import them by uncommenting one of the following lines and executing it. Once finished, comment that like again and try importing the libraries again.
Libraries are a collection of previously programmed functionalities and are useful to save time. In this example, we are going to execute a code and analyze the results.

In [ ]:
# pip --trusted-host=pypi.python.org --trusted-host=pypi.org --trusted-host=files.pythonhosted.org install pandas --user
# pip --trusted-host=pypi.python.org --trusted-host=pypi.org --trusted-host=files.pythonhosted.org install numpy --user
# pip --trusted-host=pypi.python.org --trusted-host=pypi.org --trusted-host=files.pythonhosted.org install matplotlib --user
# pip --trusted-host=pypi.python.org --trusted-host=pypi.org --trusted-host=files.pythonhosted.org install plotly --user

In [ ]:
# Import libraries
import pandas as pd # Dataframe (table) handling
import numpy as np # Numeric python, functions for calculations
import matplotlib.pyplot as plt # graphs
import plotly.express as px # more graphs
import plotly.graph_objects as go # and more graphs

# Directory path where the data is stored
path = 'https://raw.githubusercontent.com/prof-apartida/data-exercises/main/'

## Exploratory data analysis (EDA)

As mentioned, data exploration is crucial to determine if the information is accurate and has a minimum amount to get deeper insights.

### Import data

In [ ]:
df = pd.read_csv(path+'GDP_DATA.csv', header=2) # Import the data, use the data from the third row (the two forst rows have no important information, is noise)

# Function to remove the commas and convert them into numbers
def remove_commas(x):
    return x.replace(',', '') if isinstance(x, str) else x

df = df.applymap(remove_commas) # apply a function to the DataFrame
df = df.apply(pd.to_numeric, downcast='integer', errors='ignore') # downcast finds the minimum viable type for each column

df.head()# to see the first 5 rows

,Country Name,Country Code,Indicator Name,Indicator Code,1960,1961,1962,1963,1964,1965,...,2012,2013,2014,2015,2016,2017,2018,2019,2020,2021
0,Aruba,ABW,GDP (current US$),NY.GDP.MKTP.CD,NaN,NaN,NaN,NaN,NaN,NaN,...,2.615084e+09,2.727933e+09,2.791061e+09,2.963128e+09,2.983799e+09,3.092179e+09,3.202235e+09,3.310056e+09,2.496648e+09,NaN
1,Africa Eastern and Southern,AFE,GDP (current US$),NY.GDP.MKTP.CD,2.129059e+10,2.180847e+10,2.370702e+10,2.821004e+10,2.611879e+10,2.968217e+10,...,9.730435e+11,9.839370e+11,1.003679e+12,9.242525e+11,8.823551e+11,1.020647e+12,9.910223e+11,9.975340e+11,9.216459e+11,1.082096e+12
2,Afghanistan,AFG,GDP (current US$),NY.GDP.MKTP.CD,5.377778e+08,5.488889e+08,5.466667e+08,7.511112e+08,8.000000e+08,1.006667e+09,...,1.990732e+10,2.014640e+10,2.049713e+10,1.913421e+10,1.811656e+10,1.875347e+10,1.805323e+10,1.879945e+10,2.011614e+10,NaN
3,Africa Western and Central,AFW,GDP (current US$),NY.GDP.MKTP.CD,1.040414e+10,1.112789e+10,1.194319e+10,1.267633e+10,1.383837e+10,1.486223e+10,...,7.275704e+11,8.207927e+11,8.649905e+11,7.607345e+11,6.905464e+11,6.837487e+11,7.416899e+11,7.945430e+11,7.844457e+11,8.358084e+11
4,Angola,AGO,GDP (current US$),NY.GDP.MKTP.CD,NaN,NaN,NaN,NaN,NaN,NaN,...,1.249982e+11,1.334016e+11,1.372444e+11,8.721929e+10,4.984049e+10,6.897276e+10,7.779294e+10,6.930910e+10,5.361907e+10,7.254699e+10


### Get some information about the dataframe (completeness, data types, column names)

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 266 entries, 0 to 265
Data columns (total 66 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country Name    266 non-null    object 
 1   Country Code    266 non-null    object 
 2   Indicator Name  266 non-null    object 
 3   Indicator Code  266 non-null    object 
 4   1960            133 non-null    float64
 5   1961            135 non-null    float64
 6   1962            137 non-null    float64
 7   1963            137 non-null    float64
 8   1964            137 non-null    float64
 9   1965            148 non-null    float64
 10  1966            151 non-null    float64
 11  1967            155 non-null    float64
 12  1968            160 non-null    float64
 13  1969            160 non-null    float64
 14  1970            169 non-null    float64
 15  1971            172 non-null    float64
 16  1972            172 non-null    float64
 17  1973            172 non-null    flo

We can see that in 1960, we have only 133 non-null values, which means that we have a completeness of 50% (266 rows). In 2014 we have 259 non-null values, which means that we have a completeness of 97.4% in that year

### Get some statistical information

In [ ]:
df.describe() # get a summary of the numerical columns

,1960,1961,1962,1963,1964,1965,1966,1967,1968,1969,...,2012,2013,2014,2015,2016,2017,2018,2019,2020,2021
count,1.330000e+02,1.350000e+02,1.370000e+02,1.370000e+02,1.370000e+02,1.480000e+02,1.510000e+02,1.550000e+02,1.600000e+02,1.600000e+02,...,2.570000e+02,2.580000e+02,2.590000e+02,2.580000e+02,2.570000e+02,2.570000e+02,2.570000e+02,2.550000e+02,2.510000e+02,2.290000e+02
mean,7.144431e+10,7.227617e+10,7.591834e+10,8.187843e+10,8.993111e+10,9.131323e+10,1.016025e+11,1.048530e+11,1.099793e+11,1.216207e+11,...,2.409844e+12,2.486600e+12,2.547072e+12,2.399580e+12,2.444096e+12,2.619181e+12,2.780139e+12,2.844702e+12,2.794879e+12,3.488195e+12
std,2.139668e+11,2.216008e+11,2.362059e+11,2.541180e+11,2.779094e+11,2.918706e+11,3.195597e+11,3.369514e+11,3.588648e+11,3.949848e+11,...,8.177023e+12,8.401499e+12,8.612101e+12,8.159025e+12,8.315368e+12,8.848379e+12,9.414787e+12,9.587406e+12,9.380763e+12,1.108532e+13
min,1.201201e+07,1.159201e+07,9.122751e+06,1.084010e+07,1.271247e+07,1.359393e+07,1.446908e+07,1.583518e+07,1.460000e+07,1.585000e+07,...,3.934562e+07,3.861749e+07,3.875969e+07,3.681166e+07,4.162950e+07,4.521766e+07,4.781829e+07,5.422315e+07,5.505471e+07,6.310096e+07
25%,5.083344e+08,5.111424e+08,5.317365e+08,5.161478e+08,5.425784e+08,5.863716e+08,6.380995e+08,6.264909e+08,6.454036e+08,6.912229e+08,...,8.709139e+09,8.748239e+09,9.169227e+09,8.589120e+09,8.595956e+09,9.252834e+09,9.880676e+09,1.083482e+10,1.033375e+10,1.401302e+10
50%,2.760747e+09,2.899655e+09,2.814319e+09,3.540403e+09,3.405333e+09,3.038595e+09,3.170500e+09,3.384063e+09,4.064739e+09,4.759106e+09,...,4.723165e+10,5.124299e+10,5.478285e+10,4.966795e+10,4.984049e+10,5.472660e+10,5.700371e+10,6.113687e+10,5.720378e+10,6.997032e+10
75%,3.115666e+10,2.963396e+10,2.906074e+10,3.368764e+10,3.097945e+10,2.848916e+10,3.024408e+10,3.076550e+10,3.420281e+10,3.741952e+10,...,5.524837e+11,5.447092e+11,5.445767e+11,5.005287e+11,5.156547e+11,5.410187e+11,5.554554e+11,5.706808e+11,6.499961e+11,1.018007e+12
max,1.392053e+12,1.448393e+12,1.550299e+12,1.671347e+12,1.829999e+12,1.993587e+12,2.163555e+12,2.302168e+12,2.484824e+12,2.740743e+12,...,7.548806e+13,7.760720e+13,7.970881e+13,7.517927e+13,7.646559e+13,8.140398e+13,8.641303e+13,8.765286e+13,8.490681e+13,9.610009e+13


In 1960, the <b>minimum GDP</b> value was 1.201201e+07 (12012010) USD, while the <b>maximum</b> value was 1.392053e+12 (1392053000000) USD.

In 2014, the <b>minimum GDP</b> value was 3.875969e+07 (38759690) USD, while the <b>maximum</b> value was 7.970881e+13 (79708810000000) USD.

#### Graph to show outliers

This is to identify the countries with GDP values that differ significantly from the rest of the sample

In [ ]:
year = '2014'
df1 = df[['Country Code','Country Name',year]]

fig = px.box(df1, y=year, hover_data=['Country Name'])
fig.show()

### Get all possible values for column named "Country Name" and "Country code", so we can analyze if we should ignore rows (contries in this case)

In [ ]:
pd.set_option('display.max_rows', None) # ask pandas to show all rows
pd.set_option('display.max_columns', None) # ask pandas to show all columns

df[['Country Name','Country Code']]

,Country Name,Country Code
0,Aruba,ABW
1,Africa Eastern and Southern,AFE
2,Afghanistan,AFG
3,Africa Western and Central,AFW
4,Angola,AGO
5,Albania,ALB
6,Andorra,AND
7,Arab World,ARB
8,United Arab Emirates,ARE
9,Argentina,ARG


In [ ]:
# Create a list excluding the non-valid "countries" (as they are not countries)
exclude = ['AFE', 'WLD','SAS','HIC','OED','PST','IBT','LMY','MIC','IBD','EAS','UMC','LTE','ECS','NAC','EAP','TEA','EUU','EMU','EAR','LMC','LCN','TLA','LAC','TEC','TSA','MEA','ECA','ARB','IDA','SSF','TSS','SSA','CEB','FCS','MNA','IDX','TMN','PRE','LDC','IDB','HPC','AFW','SST','LIC','OSS','CSS']

df_new = df[~df['Country Code'].isin(exclude)] # Create a dataframe that does not include the aforementioned codes
df_new.describe()

fig = px.box(df_new, y="1960", hover_data=['Country Name'])
fig.show()

## Data analysis
### Descriptive analytics: graphs
#### Bar graphs

In [ ]:
# Graph the GDP values by country of the year 2014
year = '2014'

fig = px.bar(df_new, x=year, y='Country Name', title="Countries' GDP")
fig.show()

fig = px.bar(df_new.sort_values(by=year, ascending=False).reset_index(drop=True), x=year, y='Country Name', title="Countries'  GDP") # Sort the table by ascending values
fig.show()

fig = px.bar(df_new.sort_values(by=year, ascending=False).head(15).reset_index(drop=True), x=year, y='Country Name', title='Top 15 countries by GDP')# Select the first 15 countries with higher GPD values
fig.show()

fig = px.bar(df_new.sort_values(by=year, ascending=False).tail(15).reset_index(drop=True), x=year, y='Country Name', title='Bottom 15 countries by GDP')# Select the first 15 countries with lower GPD values
fig.show()

#### Pie charts

World GDP contribution by country

In [ ]:
year = '2014'
fig = px.pie(df_new.sort_values(by=year, ascending=False).reset_index(drop=True), values=year, names='Country Name', title='Top countries by GDP in '+year)
fig.show()

# Pie chart 2
fig = px.pie(df_new.sort_values(by=year, ascending=False).head(5).reset_index(drop=True), values=year, names='Country Name', title='Pie chart 2: Top 5 countries by GDP in '+year)
fig.show()

# Pie chart 3
year = '1975'
fig = px.pie(df_new.sort_values(by=year, ascending=False).head(5).reset_index(drop=True), values=year, names='Country Name', title='Pie chart 3: Top 5 countries by GDP in '+year)
fig.show()

#### World map

In [ ]:
# Create a map to spatially illustrate the GDP values
year = '2014'

fig = px.choropleth(df_new, locations='Country Code', color=year, hover_data=['Country Name'], # hover is when you place the mouse over the map
                    color_continuous_scale='Reds',  # Define the color pallete
                    color_continuous_midpoint=np.mean(df_new[year]), # the midpoint color at the mean value
                    range_color=(np.min(df_new[year]),np.max(df_new[year])), # range of intensity from the minimum to the maximum valur
                    title="GDP in " + year + " by country" )
fig.show()


fig = px.choropleth(df_new, locations='Country Code', color=year, hover_data=['Country Name'], # hover is when you place the mouse over the map
                    color_continuous_scale='hot',  # Define the color pallete
                    color_continuous_midpoint=np.mean(df_new[year]), # the midpoint color at the mean value
                    range_color=(np.min(df_new[year]),np.max(df_new[year])), # range of intensity from the minimum to the maximum valur
                    title="GDP in " + year + " by country" )
fig.show()

In [ ]:
fig = px.colors.sequential.swatches_continuous()
fig.show()

<b> Table transposition: change the data from rows, to columns and viceversa </b><br>

We want to obtain data by country, and in the other table, the columns are by year, so we proceed to transpose the table.

In [ ]:
df_t = df_new.transpose() # transpose the table
df_t = df_t.reset_index() # reset the index
df_t.columns = df_t.iloc[1] # set the row with the country codes as column names, we select the country codes, so we have no problen on typing the name
df_t = df_t.drop(df_t.index[0:4]) # remove the first rows
df_t = df_t.rename(columns={'Country Code': 'Year'}) # Rename the column
df_t = df_t.reset_index(drop=True) # reset index and create a new one (drop=True)

df_t.head()

1,Year,ABW,AFG,AGO,ALB,AND,ARE,ARG,ARM,ASM,ATG,AUS,AUT,AZE,BDI,BEL,BEN,BFA,BGD,BGR,BHR,BHS,BIH,BLR,BLZ,BMU,BOL,BRA,BRB,BRN,BTN,BWA,CAF,CAN,CHE,CHI,CHL,CHN,CIV,CMR,COD,COG,COL,COM,CPV,CRI,CUB,CUW,CYM,CYP,CZE,DEU,DJI,DMA,DNK,DOM,DZA,ECU,EGY,ERI,ESP,EST,ETH,FIN,FJI,FRA,FRO,FSM,GAB,GBR,GEO,GHA,GIB,GIN,GMB,GNB,GNQ,GRC,GRD,GRL,GTM,GUM,GUY,HKG,HND,HRV,HTI,HUN,IDN,IMN,IND,INX,IRL,IRN,IRQ,ISL,ISR,ITA,JAM,JOR,JPN,KAZ,KEN,KGZ,KHM,KIR,KNA,KOR,KWT,LAO,LBN,LBR,LBY,LCA,LIE,LKA,LSO,LTU,LUX,LVA,MAC,MAF,MAR,MCO,MDA,MDG,MDV,MEX,MHL,MKD,MLI,MLT,MMR,MNE,MNG,MNP,MOZ,MRT,MUS,MWI,MYS,NAM,NCL,NER,NGA,NIC,NLD,NOR,NPL,NRU,NZL,OMN,PAK,PAN,PER,PHL,PLW,PNG,POL,PRI,PRK,PRT,PRY,PSE,PSS,PYF,QAT,ROU,RUS,RWA,SAU,SDN,SEN,SGP,SLB,SLE,SLV,SMR,SOM,SRB,SSD,STP,SUR,SVK,SVN,SWE,SWZ,SXM,SYC,SYR,TCA,TCD,TGO,THA,TJK,TKM,TLS,TON,TTO,TUN,TUR,TUV,TZA,UGA,UKR,URY,USA,UZB,VCT,VEN,VGB,VIR,VNM,VUT,WSM,XKX,YEM,ZAF,ZMB,ZWE
0,1960,NaN,537777811.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,18606786874.0,6592693841.0,NaN,195999990.0,11658722591.0,226195579.0,330442817.0,4274893913.0,NaN,NaN,169803922.0,NaN,NaN,28071889.0,84466654.0,373879364.0,17030465539.0,NaN,NaN,NaN,30412309.0,112155599.0,40461721693.0,9522746719.0,NaN,4110000000.0,59716467625.0,546203562.0,614206071.0,3359404118.0,131731863.0,4031152977.0,NaN,NaN,507513830.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,672399700.0,2723593385.0,2069465326.0,NaN,NaN,12072126075.0,NaN,NaN,5224102196.0,112328422.0,62225478001.0,NaN,NaN,141468978.0,73233967692.0,NaN,1217086835.0,NaN,NaN,NaN,NaN,NaN,4335186017.0,NaN,NaN,1043599900.0,NaN,170215248.0,1320796652.0,335650000.0,NaN,273187200.0,NaN,NaN,NaN,37029883876.0,NaN,1939329775.0,4199134390.0,1684121535.0,248434097.0,2598500000.0,40385288344.0,699050679.0,NaN,44307342950.0,NaN,791265459.0,NaN,637142866.0,NaN,12366564.0,3958190759.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1409873950.0,34579308.0,NaN,703925706.0,NaN,NaN,NaN,2037150716.0,NaN,NaN,673081724.0,NaN,13040000000.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,162956741.0,1916241997.0,NaN,NaN,449526873.0,4196092258.0,223854667.0,12276734172.0,5163271598.0,508334414.0,NaN,5485854792.0,NaN,3749265015.0,537147100.0,2571908062.0,7515886651.0,NaN,230496033.0,NaN,1691900000.0,NaN,3193200404.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,119000024.0,NaN,1307333333.0,1003692375.0,704756305.0,NaN,322009472.0,NaN,NaN,180459937.0,NaN,NaN,NaN,99650000.0,NaN,NaN,15822585034.0,35076158.0,NaN,12012012.0,857704413.0,NaN,313582729.0,121128073.0,2760747472.0,NaN,NaN,NaN,NaN,535670128.0,NaN,13983572895.0,NaN,NaN,423008386.0,NaN,1242289239.0,543300000000.0,NaN,13066558.0,7779090909.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,8748596504.0,713000000.0,1052990400.0
1,1961,NaN,548888896.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,19683055213.0,7311749633.0,NaN,202999992.0,12400145222.0,235668222.0,350247237.0,4817580184.0,NaN,NaN,190098039.0,NaN,NaN,29964371.0,89249987.0,406684586.0,17275940449.0,NaN,NaN,NaN,32902337.0,123134584.0,40934952064.0,10712712465.0,NaN,4609727273.0,50056868958.0,618245639.0,652777614.0,3086746857.0,151675739.0,4540447761.0,NaN,NaN,490325182.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,654100200.0,2434727330.0,1753850417.0,NaN,NaN,13834300571.0,NaN,NaN,5921659485.0,116987785.0,67461644222.0,NaN,NaN,167637908.0,77741965703.0,NaN,1302521008.0,NaN,NaN,NaN,NaN,NaN,4961400439.0,NaN,NaN,1076699900.0,NaN,185848451.0,1383681651.0,356200000.0,NaN,271066000.0,NaN,NaN,NaN,39232435784.0,NaN,2088012282.0,4426949095.0,1831700364.0,253885656.0,3138500000.0,44842760293.0,748028839.0,NaN,53508617739.0,NaN,792959472.0,NaN,642857134.0,NaN,12483229.0,2417558289.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1444327731.0,35699286.0,NaN,704145671.0,NaN,NaN,NaN,2025689537.0,NaN,NaN,699161944.0,NaN,14160000000.0,NaN,NaN,NaN,NaN,584351384.0,NaN,NaN,NaN,NaN,159213140.0,NaN,174576508.0,1901868548.0,NaN,NaN,485785235.0,4467200336.0,240524723.0,13493833740.0,5632460937.0,531959562.0,NaN,5670064168.0,NaN,4118647627.0,599026300.0,2899654840.0,8171186335.0,NaN,244832035.0,NaN,1865100000.0,NaN,3417516639.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,122000016.0,NaN,1419

#### Line charts

In [ ]:
# Let's see the total GDP progress of Qatar
country = 'QAT'

fig = go.Figure()
fig.add_trace(go.Scatter(x=df_t['Year'], y=df_t[country], mode='lines', name=country))
fig.update_layout(title=country + " GDP progress")
fig.show()

In [ ]:
# Now, let's compare it with USA
country1 = 'QAT'
country2 = 'JPN'

fig = go.Figure()
fig.add_trace(go.Scatter(x=df_t['Year'], y=df_t[country1], mode='lines', name=country1))
fig.add_trace(go.Scatter(x=df_t['Year'], y=df_t[country2], mode='lines', name=country2))
fig.update_layout(title="GDP comparison")
fig.show()